sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -2612,6 +2612,11 @@ object Sequence {
       val stepDays = step.days
       val stepMicros = step.microseconds
 
+      if (scale == MICROS_PER_DAY && stepMonths == 0 && stepDays == 0) {
+        throw new IllegalArgumentException(
```

Comment:
@TJX2014 can we just use `require` instead?

Reply:
I asked to throw an exception explicitly, to match the codegen code.

Reply:
I am both ok, thank you for your attention.  :-)